### ISSUE TITLE
pipcook should looks like a community  product?

### ISSUE THREAD

**Issue by Txiaozhe** [alibaba/pipcook#65]
I means that pipcook contains many modules in its path（package）like app/cli/core and board clint/server，do not you think it's confusing？That make pipcook looks like a aggregate of many libs.
OK I would told my opinion：
For example，pipcook-board-server，a server for board contains few apis but build from egg，as every one knows that egg is an awesome framework for engineering in back end but may put fine timber to petty use，we could streamlining the board server use a lightweight framework. 
In short，I proposal we need to simplify pipcook structure by using lightweight framework，remove redundancy files，planing development mode for plugin，etc.:)

**Comment by yorkie** [comment]
See https://github.com/alibaba/pipcook/pull/52 and https://github.com/alibaba/pipcook/pull/46, we have finished some related works as you described before.

> In short，I proposal we need to simplify pipcook structure by using lightweight framework，remove redundancy files，planing development mode for plugin

Could you please give more details so that we can have more to talk about :)

**Comment by Txiaozhe** [comment]
In my opinion, I hope pipcook can become a powerful, structured and easy to expand community product, and there is a community called pipcook.

So I have the following suggestions:

1. Separate the pipcook project from the alibaba organization and establish an organization called pipbook or pipbookjs, pipbook is the main project of pipbook organization.

2. Sort out the directory structure of pipcook. I think it's a bit overqualified to use eggjs to build pipcook-board server. Clear some redundant files and dependencies to make the structure clearer.

3. Put the plugins of pipbook independently in the pipbook organization, and make them independent projects. Like the plugins of eggjs, even if there are 10000 plugins, they will not be hard to maintain.

Finally, on the one hand, it can be operated and promoted as an independent community product, and on the other hand, it can also be more flexible to manage and promote the development and use of pipbook plugin

英文水平有限，怕不能正确表达我的观点，所以用中文写一遍 :)
在我的想法里，我希望 pipcook 成为一个强大、结构清晰、易扩展的社区产品，并且有一个社区叫做 pipcook
所以我有以下几个建议：
1. 把 pipcook 项目从 alibaba 组织中脱离出来，成立一个组织叫做 pipcook 或 pipcookjs，pipcook 作为 pipcook 组织的主要项目
2. 整理 pipcook 的目录结构，我觉得用 eggjs 来实现 board server 有点大材小用，把一些冗余的文件和依赖清除，使结构更加清晰
3. 把 pipcook 的 plugin 独立出来放在 pipcook 组织里，使之成为独立项目，像 eggjs 的 plugin 那样，这样做哪怕有一万个 plugin 也能很好地维护

最后，我觉得这样做的好处，一方面可以作为一个独立的社区产品来运营和推广，另外把 plugin 独立出来也可以更灵活地去管理和推广 pipcook plugin 的开发与应用。

**Comment by utkobe** [comment]
@Txiaozhe 

Thanks a lot for the suggesions. That will be very useful.

1. According to Alibaba open-source regulations, currently we could not seperate our project from organazation. But we will try to operate pipcook as an independent product (organization) and try to build our pipcook communitry.

2. For eggjs staff, I think that makes sense. I will clean the redundant and unuseful files and codes generated by eggjs. We are not sure for now if we will have more and complicated features so we will have further discussion about if we need to use Eggjs.

3.  it's a good idea to seperate plugins and build a plugin ecosystem.  With regard to issue #17 , we have a plan for this ecosystem. The plugins will be categorized into built-in plugin (maintained by our repo), community plugin (maintained by third-party developer and their own repos) and private plugin (matained by by private organization or company itself).

Hopefully we can work together to build a powerful community. :)

**Comment by Txiaozhe** [comment]
Thanks for your sincere reply. I'd be happy to   and do my best :)

**Comment by yorkie** [comment]
@Txiaozhe closing because this is not to track anything else, and you could go #30 to see our v1 roadmap, then contribute some that you think comfortable and discussion is also welcome.
